tests/derive.rs

```diff
@@ -231,3 +231,42 @@ fn recursive_and_empty_input() {
 
     let _ = Nat5::arbitrary(&mut Unstructured::new(&[]));
 }
+
+#[test]
+fn test_field_attributes() {
+    // A type that DOES NOT implement Arbitrary
+    #[derive(Debug)]
+    struct Weight(u8);
+
+    #[derive(Debug, Arbitrary)]
+    struct Parcel {
+        #[arbitrary(with = "arbitrary_weight")]
```

Comment:
Can we also add a test where users define a closure for the custom arbitrary function?

For example:

```rust
#[arbitrary(with = |u| u.int_in_range(0..=10))]
```

I could see the generated code needing to wrap parens or put the function in a temporary variable to avoid compile errors in the generated code with this kind of macro input.

Reply:
Addressed in https://github.com/rust-fuzz/arbitrary/pull/129/commits/727ab88363108e041243375c5ed2e31389a78d32

Note, the compiler require a type hint for `u`.
So 
```rust
#[arbitrary(with = |u| u.int_in_range(0..=10))]
```
does not work,
but 
```rust
#[arbitrary(with = |u: &mut Unstructured| u.int_in_range(0..=10))]
```
does.

There is a chance to make it work without specifying the type for `u`,  but it would require some dancing around to help the compiler to derive  the type (e.g. wrapping a closure into a helper function, etc.). I think it will be too much effort and mess (in the generated code) for a very little value.